File: transpiler/src/test/java/com/here/ivi/api/platform/baseapi/BaseApiGeneratorSuiteTest.java
/*
 * Copyright (C) 2017 HERE Global B.V. and its affiliate(s). All rights reserved.
 *
 * This software, including documentation, is protected by copyright controlled by
 * HERE Global B.V. All rights are reserved. Copying, including reproducing, storing,
 * adapting or translating, any or all of this material requires the prior written
 * consent of HERE Global B.V. This material also contains confidential information,
 * which may not be disclosed to others without prior written consent of HERE Global B.V.
 *
 */

package com.here.ivi.api.platform.baseapi;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertNotNull;

import com.here.ivi.api.cli.OptionReader;
import com.here.ivi.api.generator.common.GeneratedFile;
import java.util.LinkedList;
import java.util.List;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.junit.runners.JUnit4;

@RunWith(JUnit4.class)
public final class BaseApiGeneratorSuiteTest {

  private final BaseApiGeneratorSuite baseApiGeneratorSuite =
      new BaseApiGeneratorSuite(OptionReader.DEFAULT_OPTIONS);
  private List<GeneratedFile> generatedFiles;

  private String getTargetFile(int i) {
    return generatedFiles.get(i).targetFile.toString();
  }

  @Test
  public void generateFilesEmptyModel() {
    generatedFiles = baseApiGeneratorSuite.generate(null, new LinkedList<>());
    assertNotNull(generatedFiles);
    final int expectedGeneratedFiles = BaseApiGeneratorSuite.ADDITIONAL_HEADERS.size();
    assertEquals(
        "Expected cpp/internal files and test generated file",
        expectedGeneratedFiles,
        generatedFiles.size());

    assertEquals("cpp/include/EnumHash.h", getTargetFile(0));
    assertEquals("cpp/include/Return.h", getTargetFile(1));
  }
}

File: gluecodium/src/test/java/com/here/gluecodium/platform/baseapi/BaseApiGeneratorSuiteTest.kt
/*
 * Copyright (C) 2016-2019 HERE Europe B.V.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 * License-Filename: LICENSE
 */

package com.here.gluecodium.platform.baseapi

import com.here.gluecodium.Gluecodium
import com.here.gluecodium.model.lime.LimeModel
import com.here.gluecodium.test.AssertHelpers.assertContains
import org.junit.Assert.assertEquals
import org.junit.Assert.assertNotNull
import org.junit.Before
import org.junit.Test
import org.junit.runner.RunWith
import org.junit.runners.JUnit4

@RunWith(JUnit4::class)
class BaseApiGeneratorSuiteTest {

    private lateinit var baseApiGeneratorSuite: BaseApiGeneratorSuite

    @Before
    fun setUp() {
        baseApiGeneratorSuite = BaseApiGeneratorSuite(Gluecodium.testOptions())
    }

    @Test
    fun generateFilesEmptyModel() {
        val generatedFiles = baseApiGeneratorSuite.generate(LimeModel(emptyMap(), emptyList()))
        assertNotNull(generatedFiles)
        // + 1 for Export.h
        val expectedGeneratedFiles = BaseApiGeneratorSuite.ADDITIONAL_HEADERS.size + 2
        assertEquals(
                "Expected cpp/internal files and test generated file",
                expectedGeneratedFiles,
                generatedFiles.size
        )

        val generatedFileNames = generatedFiles.map { it.targetFile.toString() }
        assertContains("cpp/include/gluecodium/Return.h", generatedFileNames)
        assertContains("cpp/include/gluecodium/Export.h", generatedFileNames)
        assertContains("cpp/include/gluecodium/Optional.h", generatedFileNames)
        assertContains("cpp/include/gluecodium/OptionalImpl.h", generatedFileNames)
        assertContains("cpp/include/gluecodium/Export.h", generatedFileNames)
    }
}
